community/server/src/main/java/org/neo4j/server/rest/security/SecurityRule.java

```diff
@@ -30,11 +30,18 @@
     /**
      * @param request The HTTP request currently under consideration.
      * @return <code>true</code> if the rule passes, <code>false</code> if the
-     *         rule fails and the request is to be rejected.
+     *         rule fails and the request is to be rejected with a "401 Unauthorized".
      */
     boolean isAuthorized(HttpServletRequest request);
 
     /**
+     * @param request The HTTP request currently under consideration.
+     * @return <code>true</code> if the rule is triggered and the request is to be rejected with a "403 Forbidden",
+     *  and <code>false</code> otherwise.
+     */
+    boolean isForbidden(HttpServletRequest request);
```

Comment:
`org.neo4j.server.rest.security.SecurityRule` is an interface that users are [expected to implement](http://neo4j.com/docs/stable/security-server.html#_server_authorization_rules), adding a new method constitutes an incompatible change.

I would say that we cannot make this change until 3.0, for two reasons:
1. We allow ourselves to make such breaking changes on major releases.
2. Most likely we will be using Java 8 for 3.0 at which point we could provide a default implementation for this method.